splunk-otel-android/src/main/java/com/splunk/rum/CurrentNetwork.java

```diff
@@ -39,22 +49,30 @@ NetworkState getState() {
         return state;
     }
 
-    @Nullable
-    String getSubType() {
-        return subType;
+    Attributes getNetworkAttributes() {
+        AttributesBuilder builder =
+                Attributes.builder().put(NET_HOST_CONNECTION_TYPE, state.getHumanName());
+
+        setIfNotNull(builder, NET_HOST_CONNECTION_SUBTYPE, subType);
+        if (haveCarrier()) {
+            setIfNotNull(builder, NET_HOST_CARRIER_NAME, carrier.getName());
+            setIfNotNull(builder, NET_HOST_CARRIER_MCC, carrier.getMobileCountryCode());
+            setIfNotNull(builder, NET_HOST_CARRIER_MNC, carrier.getMobileNetworkCode());
+            setIfNotNull(builder, NET_HOST_CARRIER_ICC, carrier.getIsoCountryCode());
+        }
+
+        return builder.build();
```

Comment:
I'm not excited about having this here. Previously, the `CurrentNetwork` class was a very simple data model object that simply pulled together the concepts of network carrier, network state, and network subtype. It is constructed simply and then exposes getters for the various specific details. It had no ties to instrumentation or to opentelemetry semantic conventions.  

I understand the need, but I think the mapping of domain-specific attributes into a telemetry system (otel or otherwise) is a separate responsibility. This responsibility should be outside of the data model class. If the `NetworkMonitor` doesn't feel like a good place, then maybe a new class that just does the mapping.

Reply:
> maybe a new class that just does the mapping.

Hm, yeah, I think I'll just do that.